src/engine/renderer/tr_cmds.cpp

```diff
@@ -418,12 +418,12 @@ void RE_SetColorGrading( int slot, qhandle_t hShader )
 		return;
 	}
 
-	if ( shader->defaultShader || !shader->stages[ 0 ] )
+	if ( shader->defaultShader || !shader->stages )
```

Comment:
Testing whether the begin pointer is null doesn't look good because it really depends on the details of how the array was allocated (e.g. `malloc(0)` may or may not be null)

Reply:
If there is no stage at all the code explicitly sets this pointer to `nullptr` and does no allocation at all.

Reply:
I know that it works with exactly how it is written now. I'm saying that it looks suspicious; provokes a "WTF" from the reader. In another part of the code you use `tess.surfaceLastStage != tess.surfaceStages` to test for emptiness. I suggest always doing it that way.

Reply:
I now test for `stage == lastStage` to test for emptiness everywhere.

Of course it works if both `stage` and `lastStage` are the same, even if those pointers are not `nullptr`.

Though, I wounder if it would be good to make sure things like `Hunk_Alloc` always return `nullptr` when `size` is zero. It may helps debugging: not only returning `nullptr` makes sure any access will crash, but also the developper could make the fair assumption nothing was allocated, helping to track down the possible problems he may face. But that can be done in another PR, and not be specific to that change.

I'm just afraid of seeing a non-`nullptr` in a debugger for something that was never allocated, and don't know what to do with it.

But comparing `stage` and `lastStage` is fine for the purpose of this PR.

Reply:
Here we have an example of how an allocation of size `0` can corrupt memory and lead to a crash, sometime in some other random places, such crash being hard to track down:

- https://github.com/Unvanquished/Unvanquished/issues/2682
- https://github.com/Unvanquished/Unvanquished/pull/2976

I don't know if it's a compiler bug though… But I wonder if it's a good idea to allocate a memory of size `0`, and what can be returned from this operation.